modules/nlu/src/backend/application/training-repo.ts

```diff
@@ -1,27 +1,29 @@
 import * as sdk from 'botpress/sdk'
+import { Transaction } from 'knex'
+import ms from 'ms'
 import { TrainingId, TrainingState, TrainingSession, I } from './typings'
 
 const TABLE_NAME = 'nlu_training_queue'
+const TRANSACTION_TIMEOUT_MS = ms('5s')
+const debug = DEBUG('nlu').sub('lifecycle')
 
-export type ITrainingRepository = I<TrainingRepository>
+export const TRANSACTION_TIMEOUT_ERROR = new Error("Transaction exceeded it's time limit")
```

Comment:
lol, first time I see this

Why not putting the msg of the error in a constant but instantiating a new error each time you throw ? I feel like there may be a state in an error instance that mutates when error is catched or played with.

I'd be more confortable with just the message being a contsant

Reply:
Old go habits, definitely gonna create a new instance every time